L/LibCURL/common.jl

```diff
@@ -0,0 +1,107 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+function build_libcurl(ARGS, name::String)
+    version = v"7.81.0"
+    hash = "ac8e1087711084548d788ef18b9b732c8de887457b81f616fc681d1044b32f98"
+
+    if name == "CURL"
+        this_is_curl_jll = true
+    elseif name == "LibCURL"
+        this_is_curl_jll = false
+    else
+        msg = "Not a valid name: $(name). Valid names are: LibCURL, CURL"
+        throw(ArgumentError(msg))
+    end
+
+    # Collection of sources required to build LibCURL
+    sources = [
+        ArchiveSource("https://curl.se/download/curl-$(version).tar.gz", hash),
+    ]
+
+    # Bash recipe for building across all platforms
+    script = raw"""
+    cd $WORKSPACE/srcdir/curl-*
+
+    # Holy crow we really configure the bitlets out of this thing
+    FLAGS=(
+        # Disable....almost everything
+        --without-ssl --without-gnutls --without-gssapi
+        --without-libidn --without-libidn2 --without-librtmp
+        --without-nss --without-polarssl
+        --without-spnego --without-libpsl --disable-ares --disable-manual
+        --disable-ldap --disable-ldaps --without-zsh-functions-dir
+        --disable-static --without-libgsasl
+
+        # A few things we actually enable
+        --with-libssh2=${prefix} --with-zlib=${prefix} --with-nghttp2=${prefix}
+        --enable-versioned-symbols
+    )
+
+
+    if [[ ${target} == *mingw* ]]; then
+        # We need to tell it where to find libssh2 on windows
+        FLAGS+=(LDFLAGS="${LDFLAGS} -L${prefix}/bin")
+
+        # We also need to tell it to link against schannel (native TLS library)
+        FLAGS+=(--with-schannel)
+    elif [[ ${target} == *darwin* ]]; then
+        # On Darwin, we need to use SecureTransport (native TLS library)
+        FLAGS+=(--with-secure-transport)
+
+        # We need to explicitly request a higher `-mmacosx-version-min` here, so that it doesn't
+        # complain about: `Symbol not found: ___isOSVersionAtLeast`
+        if [[ "${target}" == aarch64* ]]; then
+            export CFLAGS=-mmacosx-version-min=11.0
+        else
+            export CFLAGS=-mmacosx-version-min=10.11
+        fi
```

Comment:
Same as https://github.com/JuliaPackaging/Yggdrasil/pull/4153#discussion_r810645708

Reply:
I've removed the aarch64 branch. I kept the x86_64 branch.